tasks/README.md

```diff
@@ -107,19 +112,39 @@ Any parameter that does not specify a type will accept any value (default type i
 
 If a parameter type accepts `null` the task runner will accept either a `null` value or the absence of the property. Task authors must accept missing properties for nullable parameters. Task authors must not differentiate between absent properties and properties with `null` values.
 
+### Options
+
+**type**: A Puppet Type string describing the type of value the parameter accepts. Default is `Any`. (rev 1)
+
+**description**: A string description of the parameter. (rev 1)
+
+## Task implementations
+
+The implementations property is used to describe different implementations for differet targets. The value is an array of implementation options objects. For each target the task is run on the Task Runner must compare the required features of the target to it's list of available features on that target and execute the first implementation where all requirements are statisfied.
+
+### Options
+
+**name**: The file name of the task file that contains the implementation of the task. This must be at the top level of the tasks directory of the module.(rev 2)
```

Comment:
In conjunction with the example you changed above, I think it'd make sense to call out the pattern we recommend to remain compatible with runners implementing revision 1.